.reek

```diff
@@ -25,13 +25,15 @@ NilCheck:
     - TotpSetupController#valid_code?
     - user_not_found?
     - SamlIdpLogoutConcern#name_id
+    - User#password_reset_profile
 TooManyConstants:
   exclude:
     - Analytics
 TooManyStatements:
   max_statements: 6
   exclude:
     - Users::PhoneConfirmationController
+    - ReactivateProfileForm#reencrypt_pii
```

Comment:
My first thought was to split it up into two parts like this:

```diff
diff --git a/app/forms/reactivate_profile_form.rb b/app/forms/reactivate_profile_form.rb
index 21b92af..aa0f838 100644
--- a/app/forms/reactivate_profile_form.rb
+++ b/app/forms/reactivate_profile_form.rb
@@ -26,10 +26,17 @@ class ReactivateProfileForm
 
   protected
 
-  def reencrypt_pii
+  def decrypt_pii
     profile = user.password_reset_profile
     pii = profile.recover_pii(recovery_code)
     user_access_key = user.unlock_user_access_key(password)
+
+    [profile, pii, user_access_key]
+  end
+
+  def reencrypt_pii
+    profile, pii, user_access_key = decrypt_pii
+
     recovery_code = profile.encrypt_pii(user_access_key, pii)
     profile.deactivation_reason = :not_deactivated
     profile.save!
```

...and it got a feature envy reek. It's a 6-line method, and I agree it's complicated, but I'm tempted to leave it because encryption is, in fact, complicated?